aws/resource_aws_athena_database.go

```diff
@@ -38,18 +38,73 @@ func resourceAwsAthenaDatabase() *schema.Resource {
 				Optional: true,
 				Default:  false,
 			},
+			"encryption_key": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"type": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
 		},
 	}
 }
 
+func getResultConfig(d *schema.ResourceData) (*athena.ResultConfiguration, error) {
```

Comment:
Since this project has a very flat package structure, we prefer to include the service name in the function name. We also refer to these generally as "expand" functions and it would be nice to reduce the signature to not use `d` directly. e.g.

```go
func expandAthenaResultConfiguration(bucket string, encryptionKeyList []interface{}) (*athena.ResultConfiguration, error) {
```